File: Graph Algorithm/python/largestComponent.py
def largest_component(graph):
    visited=[]
    largest=0

    for i in graph:
        re=explore(graph,i,visited)
        if largest<re:
            largest=re

    return largest

def explore(graph,current,visited):
    if current in visited: return 0
    visited.append(current)

    count=1
    for i in graph[current]:
        count+=explore(graph,i,visited)

    return count

# print(largest_component({}))
print(largest_component({
  3: [],
  4: [6],
  6: [4, 5, 7, 8],
  8: [6],
  7: [6],
  5: [6],
  1: [2],
  2: [1]
}))

File: Graph Algorithm/c++/largestComponent.cpp
#include<bits/stdc++.h>
using namespace std;

int explore(std::unordered_map<int, std::vector<int>> graph, int current, unordered_map<int,bool> &visited)
{
    if (visited.find(current)!=visited.end())
        return 0;
    visited[current]=true;

    int c=1;
    vector<int>::iterator i;
    for (i=graph[current].begin();i!=graph[current].end();i++)
        c+=explore(graph,*i,visited);

    return c;
}

int largestComponent(std::unordered_map<int, std::vector<int>> graph)
{
    unordered_map<int,bool>visited;
    int largest=0;

    for (auto i:graph)
    {
        int re=explore(graph,i.first,visited);
        largest=max(largest,re);
    }
    return largest;
}

int main()
{
    std::unordered_map<int, std::vector<int>> graph {
      { 1, {2} },
      { 2, {1, 8} },
      { 6, {7} },
      { 9, {8} },
      { 7, {6, 8} },
      { 8, {9, 7, 2} }
    };
    cout<<largestComponent(graph);
}
